src/huggingface_hub/utils/_errors.py

```diff
@@ -257,13 +257,18 @@ def hf_raise_for_status(
             raise EntryNotFoundError(message, response) from e
 
         elif error_code == "RepoNotFound" or response.status_code == 401:
+            # 401 is misleading as it is returned for:
+            #    - private and gated repos if user is not authenticated
+            #    - missing repos
+            # => for now, we process them as `RepoNotFound` anyway.
+            # See https://gist.github.com/Wauplin/46c27ad266b15998ce56a6603796f0b9
```

Comment:
@huggingface/moon-landing-back to check the accuracy of this comment

Reply:
Very much related to this comment: https://github.com/huggingface/moon-landing/issues/3903#issuecomment-1320137948